spd/models/components.py

```diff
@@ -161,10 +160,68 @@ def forward(self, x: Float[Tensor, "batch pos"]) -> Float[Tensor, "batch pos emb
         # From https://github.com/pytorch/pytorch/blob/main/torch/_decomp/decompositions.py#L1211
         component_acts = self.V[x]  # (batch pos C)
 
-        if self.mask is not None:
-            component_acts *= self.mask
+        if mask_BxC is not None:
+            component_acts *= mask_BxC
 
         out = einops.einsum(
             component_acts, self.U, "batch pos C, ... C embedding_dim -> batch pos embedding_dim"
         )
         return out
+
+
+# TODO(oli) make this the only public class here
+class ReplacedComponent(nn.Module):
```

Comment:
I don't like the name "ReplacedComponent". I prefer "ReplacementComponents" but even then I don't really like that either/

The class just chooses which of two nn.Modules to run. Maybe ModuleRouter? ModuleOrComponents (ew)? 

After saying this, I realise that I don't really like the singular in LinearComponent and EmbeddingComponent. Maybe they should just be pluralised. With singular, it makes me think that it's not necessarily the same components that we talk about in SPD, but just an object/module. If you feel inspired, pluralising these would be great (or make an issue).

Reply:
My current priority order with no very strong preferences
- ModuleOrComponents (weird but explicit)
- ReplacedModule
- RouterModule